### DISCUSSION TITLE
Resizing and cropping 8.5x11 PDF

### DISCUSSION THREAD

**Discussion by milkman0007** [ImageMagick/ImageMagick#6912] [Development]
### ImageMagick version

7

### Operating system, version and so on

Win

### Description

Hi All,
I am a product manager for a thermal label manufacturer. So forgive my lack of technical knowledge. I am trying to solve a problem a lot of my customers are having and before I go to our engineering department I want to get a better understanding of the options and receive some feedback from the community.

Here is the problem I am trying to solve.
I have customers that want to print shipping labels from their label printer from sites that do not support thermal printing formats. The best example is like an eBay return label. It is formatted to be printed out of a desktop 8.5x11 printer and I want to crop, rotate, and resize save as a PNG and print it off from a 4x6 label printer. I am looking to do this printer side if possible. But I do not think ImageMagick will run on our printer as we use an RTOS. But I am guessing we can have an option to send the PDF to a remote server to be processed then back down to the printer to be printed.
The PDF files are all very small under 1MB.

Also, ImageMagick might be an overkill for what we are doing if there are some other suggestions on how this might be done printer side that would be great.

Also, I am not sure how resource-heavy of a task my job would be. I am assuming it's something light.
Below is an example of a label that goes from 8.5x11 to a 4x6.

![fedex](https://github.com/ImageMagick/ImageMagick/assets/151421115/d5944321-e4cd-42f6-82fb-dca3ad501210)

**Comment by snibgo** [comment]
The PDF page, or just the portion you want, may contain vector graphics or raster graphics or both. ("Vector" means the image is stored as many lines, either straight or curved. "Raster" means the image is stored as pixels.)

If the portion you want is a raster image, I suggest using pdfimages to extract it. Then you can use ImageMagick to rotate and resize, and save as a PNG.

If the portion is vector, or both vector and raster, you can use ImageMagick to rasterize the entire PDF page, then crop, rotate and resize. IM delegates the rasterizing to Ghostscript.

I suppose your 4x6 inch printer has a natural resolution. Do you know what that is? If it is, say, 600 dpi, then you would want the final image to be 2400x3600 pixels. You can work back from there to figure out what dpi the PDF should be rasterized at.

**Reply by milkman0007** [reply]
The printer is a 203 DPI. Would ImageMagick run on a 180 mz CPU ARM ? I was doing some research and I found MUPDF which seemed pretty simple for hardware requirements wise.  Also the PDFs I would be dealing with are all under 1MB most are 300KB

**Comment by snibgo** [comment]
ImageMagick (IM) can be built for many processors, including ARM. What is your operating system? IM is available pre-built for Windows on ARM processors; see https://imagemagick.org/script/download.php#windows . Otherwise, you would need to build from source: see https://imagemagick.org/script/install-source.php .

If you want IM to read PDF files, you also need Ghostscript.

A 180 MHz CPU is about 1/20 of the speed of a current PC, so don't expect fast performance.
